addons/pos_viva_wallet/models/pos_payment_method.py

```diff
@@ -0,0 +1,207 @@
+            }
+        session_id, pos_session_id = data_webhook.get('MerchantTrns', '').split("/") # Split to retrieve pos_sessions_id
+        try:
+            endpoint = self.viva_wallet_api_get_endpoint() + '/ecr/v1/sessions/' + session_id
+            resp = requests.get(endpoint, headers=headers, timeout=TIMEOUT)
+        except requests.exceptions.RequestException:
+            _logger.exception("Failed to call viva_wallet_bearer_token endpoint")
+            raise UserError(_("There are some issues between us and Viva Wallet, try again later. (retrieve_session_id call)"))
+
+        if resp.text and json.loads(resp.text).get('detail') == 'Could not validate credentials':
+            self._bearer_token()
+            self.retrieve_session_id(data_webhook)
+        elif resp.status_code == 202:
+            self.retrieve_session_id(data_webhook)
+        elif resp.status_code == 200:
+            data = json.loads(resp.text)
+            if data.get('success'):
+                data.update({'pos_session_id': pos_session_id})
+                data.update({'data_webhook': data_webhook})
+                self.viva_wallet_latest_response = data
+                self.send_notification(data)
+        else:
+            raise UserError(_("There are some issues between us and Viva Wallet, try again later. (retrieve_session_id response)"))
+
+    def send_notification(self, data):
+        # Send a notification to the point of sale channel to indicate that the transaction are finish
+        pos_session_sudo = request.env["pos.session"].sudo().browse(int(data.get('pos_session_id', False)))
+        if pos_session_sudo:
+            self.env['bus.bus']._sendone(pos_session_sudo._get_bus_channel_name(), 'VIVA_WALLET_LATEST_RESPONSE', pos_session_sudo.config_id.id)
+
+    def viva_wallet_send_payment_cancel(self, data):
+        if not self.env.user.has_group('point_of_sale.group_pos_user'):
+            raise AccessError(_("Do not have access to fetch token from Viva Wallet"))
+
+        headers = {
+                'Authorization': 'Bearer ' + self.viva_wallet_bearer_token,
+                'Content-Type': 'application/json'
+            }
+        session_id = data.get('sessionId')
+        cash_register_id = data.get('cashRegisterId')
+        endpoint = f'{self.viva_wallet_api_get_endpoint()}/ecr/v1/sessions/{session_id}?cashRegisterId={cash_register_id}'
+
+        try:
+            resp = requests.delete(endpoint, headers=headers, timeout=TIMEOUT)
+        except requests.exceptions.RequestException:
+            _logger.exception("Failed to call viva_wallet_send_payment_cancel endpoint")
+            return {'error': _("There are some issues between us and Viva Wallet, try again later. (viva_wallet_send_payment_cancel)")}
+
+        if resp.text and json.loads(resp.text).get('detail') == 'Could not validate credentials':
+            self._bearer_token()
+            return self.viva_wallet_send_payment_request(data)
```

Comment:
Why are we sending a payment request when the cancel request failed with invalid credentials? I think this is a mistake. Maybe we want to retry the cancel request?